frontend/src/metabase-lib/parameters/utils/parameter-fields.unit.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { createMockField } from "metabase-types/api/mocks";
+import { createMockMetadata } from "__support__/metadata";
+import { createMockUiParameter } from "metabase-lib/parameters/mock";
+import type Field from "metabase-lib/metadata/Field";
+import { hasFields } from "./parameter-fields";
+
+describe("parameters/utils/parameter-fields", () => {
+  it("`hasFields` should return true if there is at least one field in fields", () => {
+    const metadata = createMockMetadata({
+      fields: [
+        createMockField({
+          id: 1,
+        }),
+      ],
+    });
+    const field = metadata.field(1) as Field;
```

Comment:
We can avoid the cast:

```suggestion
    const field = checkNotNull(metadata.field(1));
```

Reply:
I tried to use `checkNotNull` and I got a `no-restricted-imports` error because it is part of metabase/core. The file name is `frontend/src/metabase/core/utils/types/types.ts`. Do you know a place where I could move it so it could be used inside metabase-lib?

Reply:
I think we should override the `no-restricted-imports` rule for unit tests ([like this](https://github.com/metabase/metabase/blob/e26e58229a76db7814cdc3db2f13042455625133/frontend/src/metabase/.eslintrc#L26-L31)) in `frontend/src/metabase-lib/.eslintrc`. But better to double-check this with @ranquild.

Reply:
These are utilities so theoretically they should be available inside metabase-lib proper anyway. The imports rule is incorrect. 

I have another solution to this, to move these utilities into a `metabase-lib` file and then approve it for use in `metabase-lib` [Move metabase/core/utils/types to metabase/lib/types](https://github.com/metabase/metabase/pull/35090)